python-client/giskard/models/shap_result.py

```diff
@@ -0,0 +1,97 @@
+from enum import Enum
+from dataclasses import dataclass
+from typing import Optional
+
+from shap import Explanation
+
+from giskard.core.core import ModelType, SupportedModelTypes
+from ..utils.analytics_collector import analytics
+from giskard.client.python_utils import warning
+
+
+class PanelNames(str, Enum):
+    CATEGORICAL = "Feature importance for categorical features"
+    NUMERICAL = "Feature importance for numerical features"
+    GENERAL = "Global feature importance"
+
+
+@dataclass
+class ShapResult:
+    explanations: Explanation
+    feature_types: Optional[dict] = None
+    model_type: Optional[ModelType] = None
+    only_highest_proba: Optional[bool] = True
+
+    def _validate_wandb_config(self):
+        if self.feature_types is None:
+            raise ValueError("The attribute 'ShapResult.feature_types' is needed for 'ShapResult.to_wandb()'.")
+        if self.only_highest_proba and self.model_type is None:
+            raise ValueError("The attribute 'ShapResult.model_type' is needed for 'ShapResult.to_wandb()'.")
+        if not self.only_highest_proba and self.model_type == SupportedModelTypes.CLASSIFICATION:
+            raise ValueError(
+                "We currently support 'ShapResult.to_wandb()' only with 'only_highest_proba == True' for "
+                "classification models."
+            )
+
+    def to_wandb(self, **kwargs) -> None:
+        """Create and log to the WandB run SHAP charts."""
+        from giskard.integrations.wandb.wandb_utils import wandb_run
+        from giskard.integrations.wandb.wandb_utils import _wandb_bar_plot, _wandb_general_bar_plot, _wandb_scatter_plot
+
+        self._validate_wandb_config()
+
+        with wandb_run(**kwargs) as run:
+            try:
+                charts = dict()
+
+                # Create general SHAP feature importance plot.
+                general_bar_plot = _wandb_general_bar_plot(self.explanations, list(self.feature_types.keys()))
+                charts.update({f"{PanelNames.GENERAL}/general_shap_bar_plot": general_bar_plot})
+
+                analytics.track(
+                    "wandb_integration:shap_result:shap_result",
+                    {
+                        "wandb_run_id": run.id,
+                        "cat_feat_cnt": len([c for c, t in self.feature_types.items() if t == "category"]),
```

Comment:
not a big issue, but you can do 1 pass over the list and build the counters instead of doing 3 iterations